pkg/apis/hubcluster/v1alpha1/types.go

```diff
@@ -0,0 +1,299 @@
+	// become stale at a later date.
+	// When a client creates or updates a MultiClusterWorkload, a secondary ACL check is performed to see if the client
+	// can "use" the referenced subject.  The "use" check is uniform across ALL clusters registered in the hub. It doesn't
+	// mean that every cluster honors this subject, but for each cluster does honor it, it is possible for this MultiClusterWorkload
+	// to be placed and created there.
+	AsUser Subject `json:"asUser"`
+}
+
+type ClusterWorkloadStatus struct {
+	// conditions describes the state of the controller's reconciliation functionality.  This is a union of the WorkloadStatus and
+	// overall conditions like...
+	//  1. SubjectHonored
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []Condition `json:"conditions,omitempty"  patchStrategy:"merge" patchMergeKey:"type"`
+
+	// workloadStatus track the deployment values and errors for particular kinds of workloads
+	// +optional
+	WorkloadStatus WorkloadStatus `json:"workloadStatus"`
+}
+
+type WorkloadStatus struct {
+	// type indicates the type of workload to put into a spoke cluster.  These should match the types of workloads.
+	// Valid values:
+	//  1. "Manifests"
+	Type string `json:"type"`
+
+	// manifestsStatus holds a metav1.List that contains items for the individual manifests that you want created in the
+	// spoke cluster.
+	ManifestsStatus *ManifestStatus `json:"manifestsStatus"`
+}
+
+type ManifestStatus struct {
+	// conditions describes the state of the controller's reconciliation functionality.  This is a union of the ItemStatuses
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []Condition `json:"conditions,omitempty"  patchStrategy:"merge" patchMergeKey:"type"`
+
+	// ItemStatuses track the deployment values by item
+	// +optional
+	ItemStatuses []ItemStatus `json:"itemStatuses"`
+}
+
+type ItemStatus struct {
+	Group     string `json:"group"`
+	Version   string `json:"version"`
+	Kind      string `json:"kind"`
+	Namespace string `json:"namespace,omitempty"`
+	Name      string `json:"name"`
+
+	// conditions describes the state of the controller's reconciliation functionality.
+	// known conditions include:
+	//  1. Pending
+	//  2. Reconciled
+	//  3. Complete - for "known" resources like deployments where we wait for a condition like available pods, indicates that
+	//                the logically known-good state has been reached.  For all other resource, this simply means that
+	//                the apply call succeeded.
+	//  4. Errored - errored indicates that REST returned a non-2xx return code OR that complete has taken too long.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []Condition `json:"conditions,omitempty"  patchStrategy:"merge" patchMergeKey:"type"`
+
+	// ItemStatuses track the deployment values by item
+	// +optional
+	ItemStatuses []ItemStatus `json:"itemStatuses"`
```

Comment:
I'm not sure I understand the need for the self-referential slice of `ItemStatus`. Can you elaborate?